cwd/editor.js

```diff
@@ -55,54 +112,57 @@ const makeNumInput = (id, div, num, ref) => {
 
 let viewItr = 0;
 const views = ['http://localhost:3006/glyphs/5d3744e584687c30ef9e5b47/', 'http://localhost:3006/glyphs/5d37452784687c30ef9e5b48/',
-'http://localhost:3006/glyphs/5d4885f18227675bc73b98ac/']
-
-views.forEach(view => {$.getJSON(view, function(data) {
-  let viewMsgItr = 0;
+  'http://localhost:3006/glyphs/5d4885f18227675bc73b98ac/'
+]
 
-  data.view.messages.forEach(element => {
+views.forEach(view => {
+  $.getJSON(view, function(data) {
+    let viewMsgItr = 0;
     viewItr++;
-    viewMsgItr++;
-    let gaugeItr = 0;
-
-    const currentDiv = document.getElementById(data.name);
-
-    makeTextInput("input-" + viewItr + "-" + viewMsgItr, currentDiv, element.text, null);
-    makeNumInput("prob-" + viewItr + "-" + viewMsgItr + "-1", currentDiv, element.probability, null);
-
-    // Create an update button and set it to post on click
-    makeButton('post-btn' + viewItr + "-" + viewMsgItr, currentDiv, "Update", null);
-
-    $('#post-btn' + viewItr + "-" + viewMsgItr).click(function(){
-      const viewNum = this.id.substring(8,9);
-      const msgNum = this.id.substring(10,11);
-
-      const prob = [];
-      prob.push($('#prob-' + (this.id.substring(8,)) + "-" + 1 ).val());
-
-      const settings = {
-        "url": views[viewNum - 1] + "gauges/0/messages/" + msgNum,
-        "type": "POST",
-        "timeout": 0,
-        "headers": {
-          "Content-Type": "application/x-www-form-urlencoded"
-        },
-        "data": {
-          "message":"\"" + $('#input-' + (this.id.substring(8,))).val() + "\"",
-          "probability": `\[${prob}\]`,
-          "pass": "\"" + $('#passIn').val() + "\""
-        }
-      };
 
-      $.ajax(settings).done(function (response) {
-        console.log(response);
+    data.view.messages.forEach(element => {
+      viewMsgItr++;
+
+      const currentDiv = document.getElementById(data.name);
+
+      makeTextInput("input-" + viewItr + "-" + viewMsgItr, currentDiv, element.text, null);
+      makeNumInput("prob-" + viewItr + "-" + viewMsgItr + "-1", currentDiv, element.probability, null);
+
+      // Create an update button and set it to post on click
+      makeButton('post-btn' + viewItr + "-" + viewMsgItr, currentDiv, "Update", null);
+
+      $('#post-btn' + viewItr + "-" + viewMsgItr).click(function() {
+        updateView(this);
       });
-    });
 
+      if (data.view.messages.indexOf(element) === data.view.messages.length - 1) {
```

Comment:
This is expensive.  Can you run `forEach` providing an `index` argument too and then compare it to `data.view.messages.length` that you can grab before the loop?

Reply:
I wrote that before reading the second half of the expression - I think all you want to do is have the `forEach` provide an index argument.